src/adapters/mc/OsConfigResource.c

```diff
@@ -378,6 +384,108 @@ static MI_Result SetDesiredObjectValueToDevice(const char* who, char* objectName
     return miResult;
 }
 
+static MI_Result SetProcedureObjectValueToDevice(const char* who, char* objectName, MI_Context* context)
+{
+    char* payloadString = NULL;
+    int payloadSize = 0;
+
+    JSON_Value* jsonValue = NULL;
+    char* serializedValue = NULL;
+
+    MI_Result miResult = MI_RESULT_OK;
+    int mpiResult = MPI_OK;
+
+    if ((NULL == objectName) || (NULL == g_procedureObjectValue))
+    {
+        LogError(context, miResult, GetLog(), "[%s] SetProcedureObjectValueToDevice called with an invalid object name and/or procedure object value", who);
+        return MI_RESULT_INVALID_PARAMETER;
+    }
+
+    if (NULL == (jsonValue = json_value_init_string(g_procedureObjectValue)))
+    {
+        miResult = MI_RESULT_FAILED;
+        mpiResult = ENOMEM;
+        LogError(context, miResult, GetLog(), "[%s] json_value_init_string('%s') failed", who, g_procedureObjectValue);
+    }
+    else if (NULL == (serializedValue = json_serialize_to_string(jsonValue)))
+    {
+        miResult = MI_RESULT_FAILED;
+        mpiResult = ENOMEM;
+        LogError(context, miResult, GetLog(), "[%s] json_serialize_to_string('%s') failed", who, g_procedureObjectValue);
+    }
+    else
+    {
+        payloadSize = (int)strlen(serializedValue);
+        if (NULL != (payloadString = malloc(payloadSize + 1)))
+        {
+            const OsConfigComponent *component = NULL;
+
+            memset(payloadString, 0, payloadSize + 1);
+            memcpy(payloadString, serializedValue, payloadSize);
+
+            for (component = g_components; NULL != component->Name; ++component)
+            {
+                if (!strcmp(component->Name, g_componentName))
+                {
+                    break;
+                }
+            }
+            if (NULL != component->MmiSet)
+            {
+                mpiResult = component->MmiSet(g_componentName, objectName, payloadString, payloadSize, GetLog());
+                LogInfo(context, GetLog(), "[%s] MmiSet(%s, %s, '%.*s', %d) returned %d",
+                    who, g_componentName, objectName, payloadSize, payloadString, payloadSize, mpiResult);
+
+            }
```

Comment:
Please do not do this. This is the role of the Modules Manager. Adapters cannot call MMI directly. Adapters can only call the MPI. 

Please remove this entire function and keep the NRP existing functions for get and set that talk to OSConfig over the MPI REST API.